src/main/java/ch/wisv/areafiftylan/controller/UserProfileRestController.java

```diff
@@ -0,0 +1,97 @@
+package ch.wisv.areafiftylan.controller;
+
+import ch.wisv.areafiftylan.dto.ProfileDTO;
+import ch.wisv.areafiftylan.model.Profile;
+import ch.wisv.areafiftylan.model.User;
+import ch.wisv.areafiftylan.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import static ch.wisv.areafiftylan.util.ResponseEntityBuilder.createResponseEntity;
+
+@RestController
+@RequestMapping("/users")
+public class UserProfileRestController {
+
+    private UserService userService;
+
+    @Autowired
+    UserProfileRestController(UserService userService) {
+        this.userService = userService;
+    }
+
+    /**
+     * Add a profile to a user. An empty profile is created when a user is created, so this method fills the existing
+     * fields
+     *
+     * @param userId The userId of the user to which the profile needs to be added
+     * @param input  A representation of the profile
+     *
+     * @return The user with the new profile
+     */
+    @PreAuthorize("@currentUserServiceImpl.canAccessUser(principal, #userId)")
+    @RequestMapping(value = "/{userId}/profile", method = RequestMethod.POST)
+    public ResponseEntity<?> addProfile(@PathVariable Long userId, @Validated @RequestBody ProfileDTO input) {
+        User user = userService.addProfile(userId, input);
+
+        // Create HttpHeaders to include the location of the newly created profile
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setLocation(
+                ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}/profile").buildAndExpand(user.getId())
+                        .toUri());
+
+        return createResponseEntity(HttpStatus.CREATED, httpHeaders,
+                "Profile succesfully created at" + httpHeaders.getLocation(), user.getProfile());
+    }
+
+    /**
+     * Add a profile to the current user. An empty profile is created when a user is created, so this method fills the
+     * existing fields
+     *
+     * @param input A representation of the profile
+     *
+     * @return The user with the new profile
+     */
+    @PreAuthorize("isAuthenticated()")
+    @RequestMapping(value = "/current/profile", method = RequestMethod.POST)
```

Comment:
This is the only method not with a path of `{userId}/profile`. Since the profile field is in the user, we might want to remove this method and instead use the `PUT` method `replaceCurrentUser`